open_event/api/events.py

```diff
@@ -122,9 +130,23 @@ def create(self, data, url):
         resource_location = url + '/' + str(new_event.id)
         return self.get(new_event.id), 201, {'Location': resource_location}
 
+    def validate(self, data, model=None):
+        # Validate Social links separately
+        social_links = data.get('social_links', [])
+
+        # Remove social links from original data
+        try:
+            del data['social_links']
+        except KeyError:
+            pass
+        data = BaseDAO.validate(self, data, model)
+        return data, social_links
+
     def update(self, event_id, data):
-        data = self.validate(data)
+        data, social_links = self.validate(data)
         payload = self.fix_payload(data)
+        for link in social_links:
+            save_to_db(link, 'Save social link')
```

Comment:
This will create duplicate social links in the event. It will be better if we do it such that first all existing social links of an are deleted and then new social links are set.